pkg/artifactory/resource/webhook/resource_artifactory_custom_webhook.go

```diff
@@ -0,0 +1,454 @@
+				"Allow values: %v", strings.Trim(strings.Join(DomainEventTypesSupported[webhookType], ", "), "[]")),
+		},
+		"handler": {
+			Type:     schema.TypeSet,
+			Required: true,
+			MinItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"url": {
+						Type:     schema.TypeString,
+						Required: true,
+						ValidateDiagFunc: validation.ToDiagFunc(
+							validation.All(
+								validation.IsURLWithHTTPorHTTPS,
+								validation.StringIsNotEmpty,
+							),
+						),
+						Description: "Specifies the URL that the Webhook invokes. This will be the URL that Artifactory will send an HTTP POST request to.",
+					},
+					"secrets": {
+						Type:     schema.TypeMap,
+						Optional: true,
+						Elem: &schema.Schema{
+							Type:             schema.TypeString,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringMatch(regexp.MustCompile("^[a-zA-Z_][a-zA-Z0-9_]*$"), "Secret name must match '^[a-zA-Z_][a-zA-Z0-9_]*$'\"")),
+						},
+						Description: "A set of sensitive values that will be injected in the request (headers and/or payload), comprise of key/value pair.",
+					},
+					"proxy": {
+						Type:     schema.TypeString,
+						Optional: true,
+						ValidateDiagFunc: validator.All(
+							validator.StringIsNotEmpty,
+							validator.StringIsNotURL,
+						),
+						Description: "Proxy key from Artifactory UI (Administration -> Proxies -> Configuration)",
+					},
+					"http_headers": {
+						Type:        schema.TypeMap,
+						Optional:    true,
+						Elem:        &schema.Schema{Type: schema.TypeString},
+						Description: "HTTP headers you wish to use to invoke the Webhook, comprise of key/value pair. Used in custom webhooks.",
+					},
+					"payload": {
+						Type:             schema.TypeString,
+						Optional:         true,
+						ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+						Description:      "This attribute is used to build the request body. Used in custom webhooks",
+					},
+				},
+			},
+		},
+	}
+}
+
+type CustomBaseParams struct {
+	Key         string          `json:"key"`
+	Description string          `json:"description"`
+	Enabled     bool            `json:"enabled"`
+	EventFilter EventFilter     `json:"event_filter"`
+	Handlers    []CustomHandler `json:"handlers"`
+}
+
+func (w CustomBaseParams) Id() string {
+	return w.Key
+}
+
+type CustomHandler struct {
+	HandlerType string         `json:"handler_type"`
+	Url         string         `json:"url"`
+	Secrets     []KeyValuePair `json:"secrets"`
+	Proxy       string         `json:"proxy"`
+	HttpHeaders []KeyValuePair `json:"http_headers"`
+	Payload     string         `json:"payload,omitempty"`
+}
+
+type SecretName struct {
+	Name string `json:"name"`
+}
+
+var packKeyValuePairNoValue = func(keyValuePairs []KeyValuePair, d *schema.ResourceData, url string) map[string]interface{} {
```

Comment:
Let's use a more descriptive/specific name since this func is specifically work with `secrets`.